### PR TITLE
feat(nx-plugin): add project prefix to plugin

### LINKED ISSUE #619: nx-plugin - export `createNodes` to have dynamic tasks
### User story

As a user of the `nx-plugin` I want to have dynamic targets to avoid setup and configuration cost.
In Nx v17 they released the crystal API and this should be leveraged to generate the dynamic targets.

### Acceptance criteria

- [x] export `createNodes` to be abel to register the plugins like:
```json
// nx.json
{
  "plugin": ["@code-pushup/nx-plugin"]
}
```
Add the following targets:

- [x] target `code-pushup` if `code-pushup.config.ts` is present in root - equal to execute `npx code-pushup` with nx defaults.  #773 
  - [x] consider `projectPrefix`
- [x] (nice to have) target `code-pushup--configuration` if `code-pushup.config.ts` is NOT present in root - equal to execute `npx g @code-pushup/nx-plugin:configuration` with nx defaults

### Implementation details

- init target - https://github.com/code-pushup/cli/tree/add-createNodes-init-target
- autorun taget - https://github.com/code-pushup/cli/tree/add-createNodes-api-to-nx-plugin

### PR DESCRIPTION
**Pull request by BioPhoton**
This PR adds the executor target as dynamic target. 

- added logic and tests
- renamed e2e test files
- added e2e tests

closes #619

### CODE REVIEW

File: e2e/nx-plugin-e2e/project.json [hunk 1]
```diff
@@ -18,6 +18,6 @@
       }
     }
   },
-  "implicitDependencies": ["nx-plugin"],
```
Comment (BioPhoton) [inline e2e/nx-plugin-e2e/project.json:21]: I reference a file from the build artefact in a string

File: e2e/nx-plugin-e2e/tests/executor-autorun.e2e.test.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import { afterEach, expect } from 'vitest';
+import { generateCodePushupConfig } from '@code-pushup/nx-plugin';
+import {
+  generateWorkspaceAndProject,
+  materializeTree,
+} from '@code-pushup/test-nx-utils';
+import { removeColorCodes } from '@code-pushup/test-utils';
+import { executeProcess } from '@code-pushup/utils';
+
+// @TODO replace with default bin after https://github.com/code-pushup/cli/issues/643
```
Comment (BioPhoton) [inline e2e/nx-plugin-e2e/tests/executor-autorun.e2e.test.ts:14]:
```suggestion
```

File: packages/nx-plugin/src/plugin/target/constants.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export const CODE_PUSHUP_CONFIG_REGEX =
+  /^code-pushup\.config.(\w*\.)*(ts|js|mjs)$/;
```
Comment (matejchalk) [inline packages/nx-plugin/src/plugin/target/constants.ts:2]:
```suggestion
  /^code-pushup\.config\.(\w*\.)*(ts|js|mjs)$/;
```

File: packages/nx-plugin/src/plugin/target/executor.target.unit.test.ts [hunk 4]
```diff
@@ -13,4 +13,13 @@ describe('createExecutorTarget', () => {
       executor: 'xyz:autorun',
     });
   });
+
+  it('should use projectPrefix if provides', () => {
```
Comment (matejchalk) [inline packages/nx-plugin/src/plugin/target/executor.target.unit.test.ts:17]:
```suggestion
  it('should use projectPrefix if provided', () => {
```

**Review by matejchalk — APPROVED** [review]

**Pull request merged**